examples_tests/02.ComputeShader/main.cpp

```diff
@@ -27,11 +38,584 @@ void main()
 		imageStore(outImage, ivec2(gl_GlobalInvocationID.xy), vec4(1, 1, 0, 1));
 	}
 })";
+#else
+const char* src = R"(#version 450
 
+// layout (local_size_x = 16, local_size_y = 16) in;
 
+layout (set = 0, binding = 0, rgba8) uniform writeonly image2D outImage;
+
+layout (set = 0, binding = 1) uniform UniformBufferObject
+{
+	float r;
+	float g;
+	float b;
+	float a;
+} ubo;
+
+void main()
+{
+	// if (all(lessThan(gl_GlobalInvocationID.xy, u_pushConstants.imgSize)))
+	{
+		// vec3 rgb = imageLoad(inImage, ivec2(gl_GlobalInvocationID.xy)).rgb;
+		
+		// imageStore(outImage, ivec2(gl_GlobalInvocationID.xy), vec4(1, 0, 1, 1));
+		imageStore(outImage, ivec2(gl_GlobalInvocationID.xy), vec4(ubo.r, ubo.g, ubo.b, ubo.a));
+	}
+})";
+#endif
+
+#define LOG(...) printf(__VA_ARGS__); printf("\n");
+class DemoEventCallback : public nbl::ui::IWindow::IEventCallback
+{
+	bool onWindowShown_impl() override
+	{
+		LOG("Window Shown");
+		return true;
+	}
+	bool onWindowHidden_impl() override
+	{
+		LOG("Window hidden");
+		return true;
+	}
+	bool onWindowMoved_impl(int32_t x, int32_t y) override
+	{
+		LOG("Window window moved to { %d, %d }", x, y);
+		return true;
+	}
+	bool onWindowResized_impl(uint32_t w, uint32_t h) override
+	{
+		LOG("Window resized to { %u, %u }", w, h);
+		return true;
+	}
+	bool onWindowMinimized_impl() override
+	{
+		LOG("Window minimized");
+		return true;
+	}
+	bool onWindowMaximized_impl() override
+	{
+		LOG("Window maximized");
+		return true;
+	}
+	void onGainedMouseFocus_impl() override
+	{
+		LOG("Window gained mouse focus");
+	}
+	void onLostMouseFocus_impl() override
+	{
+		LOG("Window lost mouse focus");
+	}
+	void onGainedKeyboardFocus_impl() override
+	{
+		LOG("Window gained keyboard focus");
+	}
+	void onLostKeyboardFocus_impl() override
+	{
+		LOG("Window lost keyboard focus");
+		windowShouldClose_Global = true;
+	}
+
+	void onMouseConnected_impl(core::smart_refctd_ptr<nbl::ui::IMouseEventChannel>&& mch) override
+	{
+		LOG("A mouse has been connected");
+	}
+	void onMouseDisconnected_impl(nbl::ui::IMouseEventChannel* mch) override
+	{
+		LOG("A mouse has been disconnected");
+	}
+	void onKeyboardConnected_impl(core::smart_refctd_ptr<nbl::ui::IKeyboardEventChannel>&& kbch) override
+	{
+		LOG("A keyboard has been connected");
+	}
+	void onKeyboardDisconnected_impl(nbl::ui::IKeyboardEventChannel* mch) override
+	{
+		LOG("A keyboard has been disconnected");
+	}
+};
```

Comment:
coordinate with latest @AnastaZIuk changes, you can use a callback from CommonAPI.h

Reply:
Resolved